Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2023 Vasily Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasily Nesterov
+-/
+import Mathlib.Analysis.Convex.Combination
+import Mathlib.LinearAlgebra.AffineSpace.Independent
+import Mathlib.Tactic.Linarith
+
+/-!
+# Radon's convexity theorem
+
+Radon's theorem on convex sets states that any affine dependent set can be partitioned into two sets
+whose convex hulls intersect.
+
+## Main results
+
+* `Radon_partition`: Radon convexity theorem
+
+## Tags
+convex hull, radon
+
+-/
+
+
+open Set Finset
+open BigOperators
+
+universe u
+
+variable {𝕜 : Type*} {E : Type u} [LinearOrderedField 𝕜] [AddCommGroup E] [Module 𝕜 E]
+
+/-- If `s` is a set of affine dependent vectors, it has subset `I` with the property that
+convex hulls of `I` and `s \ I` intersect. -/
+theorem Radon_partition (s : Set E)
+(h : ¬AffineIndependent 𝕜 ((↑) : s → E)) : ∃ (I : Set E), (I ⊆ s) ∧
+(Set.Nonempty ((convexHull 𝕜 I) ∩ (convexHull 𝕜 (s \ I)))) := by
```

Comment:
I would suggest formulating the theorem for a *family* of points instead of a set of points: it should be both easier to use and easier to prove. It would look like this:
```lean
theorem radon_partition {ι : Type*} {f : ι → E}
    (h : ¬AffineIndependent 𝕜 f) : ∃ (I : Set ι),
    (Set.Nonempty ((convexHull 𝕜 (f '' I)) ∩ (convexHull 𝕜 (f '' Iᶜ)))) := by
  sorry
```

Reply:
Actually I thought a bit more about it and it's not that clear that it would be easier to use. I would still say to state that version and deduce the current one from it, but I don't have a strong opinion so do as you wish.

Reply:
Thanks! I did as you suggest, and the proof really got a little easier.